### ISSUE TITLE
Create an upset mode

### ISSUE THREAD

**Issue by reconman** [synkarius/saltbot#203]
When you're stuck in the mines, aggressive betting and the multiplier don't help much. If you bet on characters that will probably win, many other people will do that too.

In the past I changed the code so the bot would bet on the one it didn't predict to win and it helped me with getting out of the mines.

That behaviour should be activatable through the configuration.

**Comment by Pauan** [comment]
I have created my own SaltyBet bot, and in my opinion the best betting mode for Matchmaking is "expected profits".

This is how it works: it looks at all of the past matches for that character, and for each match it calculates the profit:

* If it won, it adds the odds for that character.

* If it lost, it subtracts 1.

The sum of the profits is the "expected profit". In other words, it's saying "if you had bet on that character for all of those past matches, this is the percentage of profit you would have gotten".

Here's three examples to illustrate:

----

Character 1 has these past matches:

```
Win (8 / 1)
Win (5 / 1)
Win (6.5 / 1)
Win (10 / 1)
Lose
Win (4 / 1)
```

In that case the expected profit is:

```
(1 / 8) + (1 / 5) + (1 / 6.5) + (1 / 10) + (-1) + (1 / 4) = -0.1711538461538461
```

So that character causes you to *lose* 17% of your money, despite it having a 5 / 6 winrate.

----

Character 2 has these past matches:

```
Win (1 / 5)
Lose
Lose
Lose
Lose
Win (1 / 10)
Lose
Win (1 / 13)
Lose
Lose
```

In that case the expected profit is:

```
(5 / 1) + (-1) + (-1) + (-1) + (-1) + (10 / 1) + (-1) + (13 / 1) + (-1) + (-1) = 21
```

So that character gets you 210% profit, despite it having a 3 / 10 winrate.

----

Character 3 has these past matches:

```
Win (2 / 1)
Win (3.5 / 1)
Win (1 / 1)
Lose
Lose
Win (3 / 1)
Win (4 / 1)
```

In that case the expected profit is:

```
(1 / 2) + (1 / 3.5) + (1 / 1) + (-1) + (-1) + (1 / 3) + (1 / 4) = 0.3690476190476189
```

So that character only gets you ~37% profit, despite it having a 5 / 7 winrate.

----

After calculating the expected profits for each character, it simply chooses the character that has the higher expected profits.

The above algorithm is self-correcting: it accounts for both winrate and odds, so it will avoid the trap of betting on a heavily favored character, but it *also* avoids the trap of always betting on upsets.

**Comment by calexil** [comment]
https://github.com/synkarius/saltbot/blob/master/dist/config.html#L25

@reconman 

heres your config link

**Comment by reconman** [comment]
Implemented with #5df5505, configurable via the extra config page
